src/layered_config_tree/main.py

```diff
@@ -225,6 +225,23 @@ def __str__(self) -> str:
         return f"{layer}: {value}"
 
 
+class ConfigIterator:
+    """
+    An iterator for a LayeredConfigTree object.
+
+    This iterator is used to iterate over the keys of a LayeredConfigTree object.
+    """
+
+    def __init__(self, config_tree: "LayeredConfigTree"):
+        self._iterator = iter(config_tree._children)
```

Comment:
Is this guaranteed to always be in the same order? Is it possible we would lose reproducibility?